django/core/signing.py

```diff
@@ -150,11 +155,23 @@ class Signer(object):
 
     def __init__(self, key=None, sep=':', salt=None):
         # Use of native strings in all versions of Python
-        self.sep = force_str(sep)
+        self.sep = sep
         self.key = key or settings.SECRET_KEY
         self.salt = force_str(salt or
             '%s.%s' % (self.__class__.__module__, self.__class__.__name__))
 
+    @property
+    def sep(self):
+        return self._sep
+
+    @sep.setter
+    def sep(self, value):
+        sep = force_str(value)
+        if _SEP_UNSAFE.match(sep):
+            warnings.warn('Unsafe Signer separator: %r (cannot be empty or consist of only A-z0-9-_=)' % sep,
```

Comment:
I agree if we wanted to add the `setter`, we should have it for `salt` as well and I don't see much benefit to the extra complexity.